cli/src/parse.rs

```diff
@@ -25,6 +41,7 @@ pub fn parse_file_at_path(
     debug: bool,
     debug_graph: bool,
     allow_cancellation: bool,
+    stats: &mut Stats,
```

Comment:
I feel like this should be an `Option`, but the borrow checker really hated that idea, so I gave up. Anyone who knows the correct combination of `Option`, `&mut`, and `Stats` is welcome to chime in.

Reply:
I think there's some overlap between the `Stats` parameter and the `bool` return value (which currently indicates that at least one error occurred).

Maybe instead, we could change this function to return `Result<Stats>`, and change the caller to do a comparison between `stats.total_parses` and `stats.successful_parses`, instead of checking a boolean return value.